### PR TITLE
Unrecoverable reset in @After of StandaloneTestBase

### PR DESCRIPTION
**Pull request by jwang47**
* Moved IntegrationTestClient into MetaClient.resetUnrecoverably() and ProgramClient.stopAll()
* Moved StandaloneContainer to cdap-standalone module
* Added PingClientTestRun.class, NamespaceClientTestRun.class, PreferencesClientTestRun.class to ClientTestsSuite (previously they weren't being run by Bamboo at all)

### CODE REVIEW

File: cdap-unit-test-standalone/src/main/java/co/cask/cdap/test/standalone/StandaloneTestBase.java [hunk 1]
```diff
@@ -76,4 +89,15 @@ public static void tearDownClass() throws Exception {
       standaloneMain = null;
     }
   }
+
+  protected ClientConfig getClientConfig() {
```
Comment (anwar6953) [inline cdap-unit-test-standalone/src/main/java/co/cask/cdap/test/standalone/StandaloneTestBase.java]: For most clientTestRuns, wouldn't we now have two clientConfigs? (one in ClientTestBase, and one here)?

Comment (anwar6953) [inline cdap-unit-test-standalone/src/main/java/co/cask/cdap/test/standalone/StandaloneTestBase.java]: Perhaps move the clientConfig from ClientTestBase to this class.

File: cdap-client-tests/src/test/java/co/cask/cdap/client/ClientTestsSuite.java [hunk 2]
```diff
@@ -29,7 +29,9 @@
 @RunWith(Suite.class)
 @Suite.SuiteClasses({ ApplicationClientTestRun.class, DatasetClientTestRun.class, MetricsClientTestRun.class,
                       MonitorClientTestRun.class, ProcedureClientTestRun.class, ProgramClientTestRun.class,
-                      QueryClientTestRun.class, StreamClientTestRun.class, ServiceClientTestRun.class })
+                      QueryClientTestRun.class, StreamClientTestRun.class, ServiceClientTestRun.class,
+                      PingClientTestRun.class, /*AdapterClientTestRun.class, */NamespaceClientTestRun.class,
```
Comment (anwar6953) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/ClientTestsSuite.java]: remove the comment? The changes in this PR still don't allow AdapterclientTestRun to be a part of the suite.

**Comment by anwar6953** [comment]
IMO, this StandaloneTestBase should be implemented by StandaloneContainer.
But the changes in the scope of this PR LGTM.

**Comment by jwang47** [comment]
https://builds.cask.co/browse/CDAP-RBT85-1

**Comment by jwang47** [comment]
Build failed since `CLIMainTest` assumes no reset between runs. https://builds.cask.co/browse/CDAP-RBT85-2

File: cdap-client-tests/src/test/java/co/cask/cdap/client/ServiceClientTestRun.java [hunk 3]
```diff
@@ -60,7 +60,7 @@ public void setUp() throws Throwable {
   }
 
   @After
-  public void tearDown() throws Throwable {
```
Comment (anwar6953) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/ServiceClientTestRun.java]: just curious - why this change?
